sklearn/tests/test_calibration.py

```diff
@@ -455,3 +461,143 @@ def test_calibration_attributes(clf, cv):
+                             estimator_name=estimator_name)
+    viz.plot()
+    assert viz.line_.get_label() == expected_label
+
+
+def test_plot_calibration_curve_error_non_binary(pyplot, data):
+    X, y = data
+    clf = DecisionTreeClassifier()
+    clf.fit(X, y)
+
+    msg = "Only binary classification is supported."
+    with pytest.raises(ValueError, match=msg):
+        plot_calibration_curve(clf, X, y)
+
+
+def test_plot_calibration_curve_no_predict_proba(pyplot, data_binary):
+    X, y = data_binary
+
+    class MyClassifier(ClassifierMixin):
+        def fit(self, X, y):
+            self.classes_ = [0, 1]
+            return self
+
+    clf = MyClassifier().fit(X, y)
+
+    msg = "Response method 'predict_proba' not defined in"
+    with pytest.raises(ValueError, match=msg):
+        plot_calibration_curve(clf, X, y)
+
+
+def test_plot_calibration_curve_not_fitted(pyplot, data_binary):
+    X, y = data_binary
+    clf = LogisticRegression()
+
+    with pytest.raises(NotFittedError):
+        plot_calibration_curve(clf, X, y)
+
+
+@pytest.mark.parametrize("n_bins", [5, 10])
+@pytest.mark.parametrize("strategy", ["uniform", "quantile"])
+@pytest.mark.parametrize("with_strings", [True, False])
+def test_plot_calibration_curve(pyplot, data_binary, n_bins, strategy,
+                                with_strings):
+    X, y = data_binary
+
+    lr = LogisticRegression().fit(X, y)
+
+    viz = plot_calibration_curve(
+        lr, X, y, n_bins=n_bins, strategy=strategy, alpha=0.8
+    )
+
+    y_prob = lr.predict_proba(X)[:, 1]
+    prob_true, prob_pred = calibration_curve(
+        y, y_prob, n_bins=n_bins, strategy=strategy
+    )
+
+    assert_allclose(viz.prob_true, prob_true)
+    assert_allclose(viz.prob_pred, prob_pred)
+    assert_allclose(viz.y_prob, y_prob)
+
+    assert viz.estimator_name == "LogisticRegression"
+
+    # cannot fail thanks to pyplot fixture
+    import matplotlib as mpl  # noqa
+    assert isinstance(viz.line_, mpl.lines.Line2D)
+    assert viz.line_.get_alpha() == 0.8
+    assert isinstance(viz.ax_, mpl.axes.Axes)
+    assert isinstance(viz.figure_, mpl.figure.Figure)
+
+    assert viz.ax_.get_xlabel() == "Mean predicted probability"
+    assert viz.ax_.get_ylabel() == "Fraction of positives"
+    assert viz.line_.get_label() == "LogisticRegression"
+
+
+@pytest.mark.parametrize(
+    "clf", [make_pipeline(StandardScaler(), LogisticRegression()),
+            make_pipeline(make_column_transformer((StandardScaler(), [0, 1])),
+                          LogisticRegression())])
+def test_plot_calibration_curve_pipeline(pyplot, data_binary, clf):
+    X, y = data_binary
+    clf.fit(X, y)
+    viz = plot_calibration_curve(clf, X, y)
+    assert clf.__class__.__name__ in viz.line_.get_label()
+    assert viz.estimator_name == clf.__class__.__name__
+
+
+def test_plot_roc_curve_estimator_name_multiple_calls(pyplot, data_binary):
+    # non-regression test checking that the `name` used when calling
```

Comment:
roc_curve -> calibration_curve

Also it's a bit strange to see a non-regression test on new feature that isn't even merged yet :p